### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will update quite a bit of the doxygen documentation for the Cpp98 API alter a few of the logging functions and other functions in the cpp98 API to better match the primary C++ API

### Proposed changes

- update cpp98 doxygen comments
- add some missing functions to the c++98 interface
- add a few missing functions to the C api
- update readme, changelog
- update the doxygen project 
- add another doc project with the md files for MSVC for easy editing

### CODE REVIEW

File: docs/introduction/timing_exercise.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+***
+4.  Federation Setup \[A:  period=1; B:  period=2,uninterruptible; C:  period=3\]
+
+    a.	Federate C Publishes at time 3,  Federate B requests time 6, what time will it be granted?`______________`
+
+    b.	If Federate B were not uninterruptible what time would it be granted? `_______________`
+***
+![alt text](../img/timing_fed_setup.png "Exercise Federate Setup")
+5.	Federation Setup \[A:  period=1; B:  period=2; C:  period=3\], Federates will send an update when they have received an update from the all other connected federates.  Federate A sends an update at time 0, what is the update sequence
+| Time        | Federate(s)     |
```
[the remark this answers is not in the document]
Comment (nightlark) [reply docs/introduction/timing_exercise.md]: From what I remember, that was a choice between having it render correctly on ReadTheDocs and something structured like a table on GitHub, or having it render as a table on GitHub and something odd on ReadTheDocs.

**Comment by nightlark** [comment]
Either one is fine. Updating the develop branch for both at around the same time would be best.

**Comment by phlptp** [comment]
@nightlark @kdheepak 
can we add [this](https://github.com/ryanfox/sphinx-markdown-tables) to get markdown tables to work in sphinx and RTD?  or is there some other issue?

**Comment by phlptp** [comment]
I am going to work on the examples some more this weekend.  Looks like there might be some issue with the install and finding the zmq library as I get some linking errors when building on windows for the examples repository.    I might also upgrade the included ZMQ to 4.3.2

**Comment by nightlark** [comment]
We'd need to make sure the format of the tables with the extension is the same as on GitHub, since tables aren't part of the Markdown spec and different implementations may use different syntax.

**Comment by nightlark** [comment]
It seems to think each of the exercises is part of a separate list.

**Comment by nightlark** [comment]
The timing exercises are looking pretty good on RTD. I think the last nit for the question formatting is the tables for the last question/answer look like they need to be indented as well.

**Comment by nightlark** [comment]
Well, the center tag works. The link to the answers broke -- might just need an extra line added after the closing center tag.

**Pull request merged**